### PR TITLE
[synthetics] Parameterize unit tests for deprecated usages

### PR DESCRIPTION
**Pull request by Drarig29**
### What and why?

We deprecated `global` and `config` in favor of `defaultTestOverrides` and `testOverrides` (#1285 and #1306), and we still want to unit test the old usage.

Instead of duplicating some unit tests, we are using `test.each()` to test the deprecated and current behaviors.

### How?

Use `test.each()` and computed object properties

### Review checklist

- [x] Feature or bugfix MUST have appropriate tests (unit, integration)

### CODE REVIEW

File: src/commands/synthetics/__tests__/run-tests-lib.test.ts [hunk 1]
```diff
@@ -85,47 +102,50 @@ describe('run-test', () => {
...
+            ...ciConfig,
+            [defaultTestOverrides]: userConfigOverride,
+            publicIds: ['aaa-aaa-aaa', 'bbb-bbb-bbb'],
+          })
+        ).rejects.toThrow()
+        expect(getTestsToTriggersMock).toHaveBeenCalledWith(
+          apiHelper,
+          expect.arrayContaining([
+            expect.objectContaining({id: 'aaa-aaa-aaa', testOverrides: userConfigOverride}),
+            expect.objectContaining({id: 'bbb-bbb-bbb', testOverrides: userConfigOverride}),
```
Comment (Drarig29) [inline src/commands/synthetics/__tests__/run-tests-lib.test.ts:136]: As you can see, we expect the object to contain `testOverrides` even in the deprecated case because of our backwards compatibility translation layer.

**Review by AntoineDona — APPROVED** [review]

File: src/commands/synthetics/__tests__/run-tests-lib.test.ts [hunk 2]
```diff
@@ -390,6 +418,11 @@ describe('run-test', () => {
       const testStream = fs.createReadStream('/dev/null')
       jest.spyOn(fs, 'createReadStream').mockReturnValue(testStream)
 
+      const {AppUploadReporter} = jest.requireActual<typeof appUploadReporterModule>('../reporters/mobile/app-upload')
+      jest
+        .spyOn(appUploadReporterModule, 'AppUploadReporter')
+        .mockImplementation(() => new AppUploadReporter({stdout: {write: jest.fn()}} as any))
+
```
Comment (teodor2312) [inline src/commands/synthetics/__tests__/run-tests-lib.test.ts:425]: Just out of curiosity why is this part needed?

**Review by teodor2312 — APPROVED** [review]
LGTM

File: src/commands/synthetics/__tests__/run-tests-lib.test.ts (hunk 2 above)
Comment (Drarig29) [reply src/commands/synthetics/__tests__/run-tests-lib.test.ts:425]: Sorry, didn't see the comment. It's unrelated - just to avoid having logs printed in the output by some reporters which aren't mocked
